RecoMET/METProducers/src/HcalNoiseInfoProducer.cc

```diff
@@ -654,174 +705,83 @@ HcalNoiseInfoProducer::filldigis(edm::Event& iEvent, const edm::EventSetup& iSet
-      for( unsigned ncheck = 1; ncheck <= nts ; ncheck++ ) {
-        bool cap_match = true; //will be set to false if at least one check fails below
-        bool adc_match = true; //will be set to false if at least one check fails below
-
-        // loop over the channel TS, this is for the later fiber in time
-        for( unsigned lidx = 0; lidx < ncheck; lidx++) {
-          // we are looping over the TS of the later fiber in time
-          // the TS of the earlier fiber starts from the end
-          unsigned eidx = nts-ncheck+lidx;
-          // if we get an invald value, this fiber has no data
-          // the check and match will fail, so the start_ts will 
-          // be decrimented
-          if( lasmon_capids[fidx][eidx] == -1 || lasmon_capids[fidx+1][lidx] == -1 ) {
-            cap_match = false;
-            adc_match = false;
-            break;
-          }
-
-          if( lasmon_capids[fidx][eidx] != lasmon_capids[fidx+1][lidx] ) {
-            cap_match = false;
-          }
-          // check the data values as well
-          if( lasmon_adcs[fidx][eidx] != lasmon_adcs[fidx+1][lidx] ) {
-            adc_match = false;
-          }
-        }
-        if( cap_match && (start_ts > latest_cap_match) ) {
-          latest_cap_match = start_ts;
-        }
-        if( cap_match && adc_match ) {
-          // end the loop and we'll take the current start_ts
-          // as the end of the data for this fiber
-          break;
-        }
-        else {
-          // if we don't have a match, then decrement the 
-          // starting TS and check again
-          start_ts--;
-        }
-      }
 
-      // now make some sanity checks on the determined overlap index
-      if( start_ts == -1 ) {
-        // if we didn't find any match, use the capID only to compare
-        if( latest_cap_match < 0 ) {
-          //this shouldn't happen, in this case use all the data from the fiber
-          start_ts = nts;
-        }
-        else {
-          // its possible that the timing of the fibers
-          // is shifted such that they do not overlap
-          // and we just want to stitch the fibers
-          // together with no removal.
-          // In this case the capIDs will match at the
-          // N-4 spot (and the ADCs will not)
-          // if this is not the case, then we just take
-          // the value of latest match
-          if( latest_cap_match == int(nts - 4) ) {
-            start_ts = nts;
-          } else {
-            start_ts = latest_cap_match;
-          }
-        }
-      }
-
-      // now store as the matching index
-      matching_idx.push_back(start_ts);
-    }
+          // loop over time slices
+          for( unsigned its = 0; its < ts_size; ++its ) {
+            // if we are on the last channel, use all the data
+            // otherwise only take the unique samples 
+            if( ( ich < (nch - 1) ) && its >= unsigned(laserMonitorSamples_) ) continue;
```

Comment:
This is protected by the range of the for loop.
However, since you are dealing with unsigned integers, subtractions should be better avoided: please use 
```
( (ich + 1) < nch ) 
```
instead